graphql/src/main/java/com/scalar/db/graphql/instrumentation/validation/ScanStartAndEndValidation.java

```diff
@@ -0,0 +1,62 @@
+package com.scalar.db.graphql.instrumentation.validation;
+
+import com.google.common.collect.Sets;
+import com.scalar.db.graphql.schema.Constants;
+import graphql.ErrorType;
+import graphql.GraphQLError;
+import graphql.GraphqlErrorBuilder;
+import graphql.execution.instrumentation.fieldvalidation.FieldAndArguments;
+import graphql.execution.instrumentation.fieldvalidation.FieldValidation;
+import graphql.execution.instrumentation.fieldvalidation.FieldValidationEnvironment;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+public class ScanStartAndEndValidation implements FieldValidation {
+  @SuppressWarnings("unchecked")
+  @Override
+  public List<GraphQLError> validateFields(FieldValidationEnvironment validationEnvironment) {
+    List<GraphQLError> errors = new ArrayList<>();
+    for (FieldAndArguments fieldAndArguments : validationEnvironment.getFields()) {
+      String fieldName = fieldAndArguments.getField().getName();
+      if (fieldName.endsWith("_scan")) {
+        Map<String, Object> scan = fieldAndArguments.getArgumentValue("scan");
+        List<Map<String, Object>> start = (List<Map<String, Object>>) scan.get("start");
+        if (start != null) {
+          start.forEach(
+              clusteringKey -> {
+                validateClusteringKey(errors, "start", clusteringKey, fieldAndArguments);
+              });
```

Comment:
Sorry, but other than the braces issue, I also adapted how the validation method reports an error to the way `ConditionalExpressionValidation` in this PR does. Now the method returns an `Optional` of an error instead of modifying the `errors` list that is passed by the argument. Can you please take another quick look?